crates/uv-resolver/src/pubgrub/priority.rs

```diff
@@ -224,6 +227,9 @@ pub(crate) enum PubGrubPriority {
     /// [`ForkUrls`].
     DirectUrl(Reverse<usize>),
 
+    /// The package is a proxy package.
```

Comment:
Can you expand this by saying something like:

> We process these first since each proxy package expands into two regular `PubGrubPackage` packages, which give us additional constraints, while not changing the priorities since the materialized packages are of the same package name.

It took me a while to get why this was safe to do.